esy/src/bin/esy.js

```diff
@@ -267,6 +269,53 @@ async function buildEjectCommand(sandboxPath) {
   );
 }
 
+const AVAILABLE_RELEASE_TYPE = ['dev', 'pack', 'bin'];
+
+async function releaseCommand(sandboxPath, _commandName, type, ...args) {
+  if (type == null) {
+    throw new Error('esy release: provide type');
+  }
+  if (AVAILABLE_RELEASE_TYPE.indexOf(type) === -1) {
+    throw new Error(
+      `esy release: invalid release type, must be one of: ${AVAILABLE_RELEASE_TYPE.join(', ')}`,
+    );
+  }
+
+  const releaseTag = type === 'bin' ? `bin-${os.platform()}` : type;
+  const outputPath = path.join(sandboxPath, '_release', releaseTag);
+
+  // Strip all dev metadata and make sure we see what npm registry would see.
+  // We use `npm pack` for that.
+  const tarFilename = await child_process.spawn('npm', ['pack'], {cwd: sandboxPath});
+  await child_process.spawn('tar', ['xzf', tarFilename]);
+  await pfs.mkdirp(path.dirname(outputPath));
+  await pfs.rmdir(outputPath);
+  await pfs.rename(path.join(sandboxPath, 'package'), outputPath);
+  await pfs.unlink(tarFilename);
+
+  // Copy esyrelease.js executable over to release package.
+  const esyReleaseCommandOrigin = require.resolve('./esyrelease.js');
```

Comment:
Does this need to be included in the released package so that we don't need to include a dependency on `esy` for packs?

Reply:
No, it's not needed as it isn't executed on client install at all. I just kept it that way (similar to `reason-cli`) before we reach sensible test coverage.